File: 0204-count-primes/0204-count-primes.java
class Solution {
    public int countPrimes(int n) {
        boolean [] isPrime = new boolean[n+1];
        for(int i=0;i<n;i++){
            isPrime[i]=true;
        }
        for(int i=2;i<Math.sqrt(n);i++){
            if(isPrime[i]==true){
                for(int j=2;j*i<n;j++){
                    isPrime[i*j]=false;
                }
            }
        }
        int count=0;
        for(int i=2;i<n;i++){
            if(isPrime[i]==true){
                count++;
            }
        }
        return count;
    }
}

File: 0204-count-primes/0204-count-primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        if(n<=2)
            return 0;
        vector<bool> ans(n+1,true);
        ans[0]=false;ans[1]=false;
        for(int i=2;i*i<n;i++){
            if(ans[i]==true){
                for(int j=2*i;j<n;j+=i){
                ans[j]=false;
                }
            }
        }
        int c=0;
        for(int i=2;i<n;i++){
            if(ans[i]==true) c++;
        }
        return c;
    }
    
};